### DISCUSSION TITLE
Implement rainfall data

### DISCUSSION THREAD

**Discussion by ns9920** [danwild/leaflet-velocity#90] [General]
Hi 

just curious to know how can we implement the rainfall data like wind 

Thankyou

**Comment by danwild** [comment]
Hi @ns9920,

This plugin is specifically for visualising velocity _vectors_ (i.e. direction + magnitude of combined u/v components).

Rainfall data is typically reported as magnitude per unit time, e.g. `mm/day`, and visualised as a raster (e.g. heatmap).

Therefore this plugin unlikely to be a practical candidate for visualising rainfall, unless I am missing something - more information?

**Reply by DrPDash** [reply]
Hey Dan, @ns9920 

I may be wrong but wanted to add some of my thoughts. 

I think its a good idea but is beyond the scope of the current set-up. This plugin aids in visualizing "instantaneous" animation, i.e., the direction of flow of wind or surface current at a given instance (feeding through a single file for a specific time). You could externally change the file/data-feed to get the next frame. I use it in one of the NOAA visualization tools for GFS data and 0.25 degree surface currents: https://www.star.nesdis.noaa.gov/socd/ov/

To show something like (movement of) rain, it will need an animation of animation, i.e., support looping over multiple files. As you rightly mentioned, I think the heat-map approach would be more appropriate here rather than this plugin.

Thanks,
PD

**Reply by danwild** [reply]
Hey @prasanjitdash,

> To show something like (movement of) rain

I think the keyword here is _movement_ - that is what this plugin is designed for. Sure, we can loop through a bunch of time slices to animate a temporal range rather than a single time, but we’re still talking about _vectors_ (i.e. not `mm/day`)

Unless @ns9920 wants to show the velocity of rain in the XY plane (and they have this data), it’s unlikely this plugin is of much use.

In my experience rainfall plots/animations (e.g. from radar) are more typically presented like this (raster imagery of scalar data):

![BOM Radar loop](http://www.stormygirl.net/images/radarloop021106.gif)

Cheers,
Dan

**Reply by brentfraser** [reply]
Interesting stuff.  I initially was going to respond with an animated GIF similar to the one @danwild  posted.  Then I imagined a couple of other scenarios:

1.  Static map + Animated Symbology: Instead of the traditional [non-animated] weather map with green polygons showing the location of rain clouds, use some kind of animation inside the polygons indicating rain (https://gifimage.net/raining-gif-transparent-6/).  Or better yet, use (misuse?) the leaflet-velocity to show smoother rain drops (basically as just a symbology tool).

2.  Animated Map + Animated Symbology: Building on the Dan's radar animation, instead of a single time-slice of vector info, have 5 or 6 time slices of vector info (say 20 minutes apart) and animate them.  Have leaflet-velocity loop between them showing how the vectors are changing.  Not sure, but this might bring a browser to it's knees.

**Reply by DrPDash** [reply]
Thanks @danwild @brentfraser for further thoughts. 

Yes, Dan, you are right, its basically `scalar` vs. `vector` and `rainfall (mm/day)` is a scalar quantity. The rain movement animation example seems the most appropriate way for quantifying and locating the rain amount.

To Brent's 2nd point, this will surely result in some lag. I actually have this externally implemented in the link I shared earlier. One way would be to have vector tiling any global data at multiple zoom levels but that's certainly a lot to ask for from the plugin that is very effective in what it does currently. Your 1st point is perhaps some thing to think about, its visually appealing even if it does not add any more information than Dan's animated GIF.

**Reply by brentfraser** [reply]
For the 2nd case, when implementing traditional radar animation, I've had good success implementing it with [e.g. 6] separate leaflet layers, loading them all, then using show/hide to animate through them.  Perhaps this approach would improve performance.  We might need to tweak leaflet-velocity to ensure it doesn't animate hidden layers (I haven't looked into this).
